default/python/AI/ShipDesignAI.py

```diff
@@ -1093,7 +1086,7 @@ def optimize_design(self, additional_parts=(), additional_hulls=(),
         elif isinstance(loc, list):
             planets = loc
         else:
-            print >> sys.stderr, "Invalid loc parameter for optimize_design(). Expected int or list but got", loc
+            warn("Invalid loc parameter for optimize_design(). Expected int or list but got %s" % loc)
```

Comment:
The data type is invalid and this should never happen, so it should be an error.